[RFC] Maintainer guidelines

BenWibking opened the pull request:
### Description
This is my request for comments (RFC) proposal to add Maintainer Guidelines to the documentation, and officially adopt them for the Quokka project.

### Related issues
N/A

### Checklist
_Before this pull request can be reviewed, all of these tasks should be completed. Denote completed tasks with an `x` inside the square brackets `[ ]` in the Markdown source below:_
- [x] I have added a description (see above).
- [ ] I have added a link to any related issues (if applicable; see above).
- [x] I have read the [Contributing Guide](https://github.com/quokka-astro/quokka/blob/development/CONTRIBUTING.md).
- [ ] I have added tests for any new physics that this PR adds to the code.
- [ ] *(For quokka-astro org members)* I have manually triggered the GPU tests with the magic comment `/azp run`.

BenWibking committed:
docs: add maintainer guidelines

BenWibking committed:
edit

BenWibking committed:
edit

markkrumholz commented:
This looks like a very good starting place. Before adopting policy changes, as opposed to code or technical changes, I think it would be good to have a real-time telecon discussion. That includes this change. So I'd like to propose that (1) we distinguish changes in collaboration policy from technical changes and make clear that the protocol you're proposing applies only to the latter, and (2) that prior to adopting this particular change we discuss it on a telecon. 

My reason for saying this is that, while github issues are a good place for technical discussion, policies are largely about culture and sociology, and I think text-based exchanges are a poor forum for culture discussion (as exemplified by, say, the entire history of the internet...).

BenWibking commented:
> This looks like a very good starting place. Before adopting policy changes, as opposed to code or technical changes, I think it would be good to have a real-time telecon discussion. That includes this change. So I'd like to propose that (1) we distinguish changes in collaboration policy from technical changes and make clear that the protocol you're proposing applies only to the latter, and (2) that prior to adopting this particular change we discuss it on a telecon.
> 
> My reason for saying this is that, while github issues are a good place for technical discussion, policies are largely about culture and sociology, and I think text-based exchanges are a poor forum for culture discussion (as exemplified by, say, the entire history of the internet...).

Totally agree. Let's discuss this on an upcoming telecon.

BenWibking commented:
HackMD documents (with commenting turned on):
* Guidelines: https://hackmd.io/@bwibking/S1mOC0PLWx
* PR Revert policy: https://hackmd.io/@bwibking/rkZDkyuLWl
* Architectural Decision Records (ADRs): https://hackmd.io/@bwibking/HJMKJyu8Wg

BenWibking committed:
Update maintainer_guidelines.md

BenWibking committed:
Update PR revert policy for clarity on serious issues

Clarify revert policy to include beta features and specify conditions for triggering a revert.

BenWibking committed:
Merge branch 'development' into BenWibking/maintainer-guidelines

markkrumholz commented:
@BenWibking, there is one unresolved comment from codex, which says that there is a bug in the `git revert` command syntax. This exceeds my git ninja skills, so can you take a look and see if that comment is correct or a hallucination, and either resolve or fix the PR depending in which?
